shopfloor_batch_automatic_creation/actions/picking_batch_auto_create.py

```diff
@@ -104,11 +104,17 @@ def _search_pickings(self, picking_types, user=None, commercial_partner=None):
         # We can't use a limit in the SQL search because the 'priority' fields
         # is sometimes empty (it seems the inverse StockPicking.priority field
         # mess up with default on stock.move), we have to sort in Python.
-        return self.env["stock.picking"].search(
+        pickings = self.env["stock.picking"].search(
             self._search_pickings_domain(
                 picking_types, user=user, commercial_partner=commercial_partner
             )
         )
+        # The selected pickings need to match the type of the first picking found.
+        first_picking_type = fields.first(pickings).picking_type_id
+        filtered_pickings = pickings.filtered(
+            lambda p: p.picking_type_id == first_picking_type
+        )
```

Comment:
Why we cannot do this:
```suggestion
        domain = self._search_pickings_domain(
            picking_types, user=user, commercial_partner=commercial_partner
        )
        first_picking = self.env["stock.picking"].search(domain, limit=1)
        first_picking_type = first_picking.picking_type_id
        domain.append(("picking_type_id", "=", first_picking_type.id))
        return self.env["stock.picking"].search(domain)
```
It's two search, but it should be quite fast and low on memory.